.github/actions/deploy-cloud-run/action.yml

```diff
@@ -0,0 +1,81 @@
+name: Deploy Cloud Run service
+description: Create a new revision and rollback if health check fails
+inputs:
+  region:
+    description: Region of the Cloud Run service
+    required: true
+  image:
+    description: Image name
+    required: true
+  service:
+    description: Name of the Cloud Run service
+    required: true
+  health_check_url:
+    description: Health check URL. Multiple URLs can be separated by line breaks. If not provided, health check will be skipped.
+
+runs:
+  using: composite
+  steps:
+    - name: Get current revision
+      id: get_revision
+      shell: bash
+      run: |
+        REV=$( \
+          gcloud run services describe "${{ inputs.service }}" \
+            --region "${{ inputs.region }}" \
+            --format="value(status.latestReadyRevisionName)" \
+        )
+        echo "Current revision: $REV"
+        echo "revision=$REV" >> "$GITHUB_OUTPUT"
+
+    - name: Deploy
+      shell: bash
+      run: |
+        gcloud run deploy "${{ inputs.service }}" \
+          --image "${{ inputs.image }}" \
+          --region "${{ inputs.region }}" \
+          --platform managed \
+          --quiet
+
+    # ensure that the traffic is updated to the latest revision always
+    - name: Update traffic
+      shell: bash
+      run: |
+        gcloud run services update-traffic "${{ inputs.service }}" \
+          --to-latest \
+          --region "${{ inputs.region }}"
+
+    # if the URL is invalid, curl itself will return an error code
+    - name: Health check
+      id: health_check
+      if: ${{ inputs.health_check_url }}
```

Comment:
We could utilize aforementioned [GH actions output to get URL](https://github.com/google-github-actions/deploy-cloudrun?tab=readme-ov-file#outputs).

But the problem is to hit `/` or `/api/ping` as we have different healthcheck path for API service and Web service.